File: backend/burning-okr/burning-okr-model/src/main/java/org/burningokr/model/okr/OkrTopicDescription.java
package org.burningokr.model.okr;

import jakarta.persistence.*;
import lombok.Data;
import org.burningokr.model.activity.Trackable;

import java.time.LocalDate;
import java.util.Collection;
import java.util.UUID;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@Data
public class OkrTopicDescription implements Trackable<Long> {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Long id;

  @Column(length = 255)
  private String name;

  private UUID initiatorId;

  @ElementCollection
  @CollectionTable(name = "okr_description_member")
  @Column(name = "okr_member_id")
  private Collection<UUID> startTeam;

  @ElementCollection
  @CollectionTable(name = "okr_description_stakeholder")
  @Column(name = "okr_stakeholder_id")
  private Collection<UUID> stakeholders;

  @Column(length = 1023)
  private String description;

  @Column(length = 1023)
  private String contributesTo;

  @Column(length = 1023)
  private String delimitation;

  private LocalDate beginning;

  @Column(length = 1023)
  private String dependencies;

  @Column(length = 1023)
  private String resources;

  @Column(length = 1023)
  private String handoverPlan;

  public OkrTopicDescription() {}

  public OkrTopicDescription(String name) {
    this.name = name;
  }
}

File: frontend/src/app/shared/model/ui/OrganizationalUnit/okr-topic-description.ts
import { TopicDescriptionId, UserId } from '../../id-types';

export class OkrTopicDescription {
  id: TopicDescriptionId;
  name: string;
  initiatorId: UserId;
  startTeam: UserId[];
  stakeholders: UserId[];
  acceptanceCriteria: string;
  contributesTo: string;
  delimitation: string;
  beginning: Date;
  dependencies: string;
  resources: string;
  handoverPlan: string;

  constructor(descriptionId: TopicDescriptionId, name: string, initiatorId: UserId, startTeam: UserId[], stakeholders: UserId[],
              acceptanceCriteria: string, contributesTo: string, delimitation: string, beginning: Date,
              dependencies: string, resources: string, handoverPlan: string) {
    this.id = descriptionId;
    this.name = name;
    this.initiatorId = initiatorId;
    this.startTeam = startTeam;
    this.stakeholders = stakeholders;
    this.acceptanceCriteria = acceptanceCriteria;
    this.contributesTo = contributesTo;
    this.delimitation = delimitation;
    this.beginning = beginning;
    this.dependencies = dependencies;
    this.resources = resources;
    this.handoverPlan = handoverPlan;
  }
}
